Panic when running ExampleGenerativeModel_GenerateContentStream test

## Code
I just change the prompt text to "count 1 to 100.". The code is copied from ExampleGenerativeModel_GenerateContentStream in the file `example_test.go` of this repository. I meet it everytime.

```go
func TestExampleGenerativeModel_GenerateContentStream(t *testing.T) {
	ctx := context.Background()
	client, err := genai.NewClient(ctx, option.WithAPIKey(os.Getenv("API_KEY")))
	if err != nil {
		log.Fatal(err)
	}
	defer client.Close()

	model := client.GenerativeModel("gemini-pro")

	iter := model.GenerateContentStream(ctx, genai.Text("count 1 to 100."))
	for {
		resp, err := iter.Next()
		if err == iterator.Done {
			break
		}
		if err != nil {
			log.Fatal(err)
		}
		printResponseV1(resp)
	}
}
```

## Test Result

```bash
1, 2, 3, 4, 5, 6
---
, 7, 8, 9, 10, 11, 12, 13, 14, 1
---
5, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 3
---
--- FAIL: TestExampleGenerativeModel_GenerateContentStream (4.49s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x104db8304]

goroutine 20 [running]:
testing.tRunner.func1.2({0x104f58140, 0x1054c4c00})
	/opt/homebrew/opt/go/libexec/src/testing/testing.go:1545 +0x1c8
testing.tRunner.func1()
	/opt/homebrew/opt/go/libexec/src/testing/testing.go:1548 +0x360
panic({0x104f58140?, 0x1054c4c00?})
	/opt/homebrew/opt/go/libexec/src/runtime/panic.go:914 +0x218
github.com/google/generative-ai-go/genai.joinContent(...)
	/Users/xxx/go/pkg/mod/github.com/google/generative-ai-go@v0.3.0/genai/client.go:298
github.com/google/generative-ai-go/genai.joinCandidateLists({0x1400006a148, 0x1, 0x1}, {0x1400006a158, 0x1, 0x105051800?})
	/Users/xxx/go/pkg/mod/github.com/google/generative-ai-go@v0.3.0/genai/client.go:271 +0x144
github.com/google/generative-ai-go/genai.joinResponses(...)
	/Users/xxx/go/pkg/mod/github.com/google/generative-ai-go@v0.3.0/genai/client.go:257
github.com/google/generative-ai-go/genai.(*GenerateContentResponseIterator).Next(0x1400009f710)
	/Users/xxx/go/pkg/mod/github.com/google/generative-ai-go@v0.3.0/genai/client.go:178 +0x228

```

--- a/genai/client.go
+++ b/genai/client.go
@@ -292,6 +292,9 @@ func joinContent(dest, src *Content) *Content {
 	if dest == nil {
 		return src
 	}
+	if src == nil {
+		return dest
+	}
 	// Assume roles are the same.
 	dest.Parts = joinParts(dest.Parts, src.Parts)
 	return dest

--- a/genai/client_test.go
+++ b/genai/client_test.go
@@ -66,7 +66,11 @@ func TestLive(t *testing.T) {
 		got := responsesString(t, iter)
 		checkMatch(t, got, `(don't|do\s+not|not capable) (have|possess|experiencing) .*(a .* needs|body|sensations|the ability)`)
 	})
-
+	t.Run("streaming-counting", func(t *testing.T) {
+		// Verify only that we don't crash. See #18.
+		iter := model.GenerateContentStream(ctx, Text("count 1 to 100."))
+		_ = responsesString(t, iter)
+	})
 	t.Run("chat", func(t *testing.T) {
 		session := model.StartChat()
 
@@ -352,6 +356,9 @@ func responseString(resp *GenerateContentResponse) string {
 
 func contentString(c *Content) string {
 	var b strings.Builder
+	if c == nil || c.Parts == nil {
+		return ""
+	}
 	for i, part := range c.Parts {
 		if i > 0 {
 			fmt.Fprintf(&b, ";")
@@ -448,27 +455,5 @@ func TestMatchString(t *testing.T) {
 		if !re.MatchString(test.in) {
 			t.Errorf("%q doesn't match %q", test.re, test.in)
 		}
-	}
-}
-
-func TestStreamCount(t *testing.T) {
-	ctx := context.Background()
-	client, err := NewClient(ctx, option.WithAPIKey(*apiKey))
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer client.Close()
-	model := client.GenerativeModel(*modelName)
-
-	iter := model.GenerateContentStream(ctx, Text("count 1 to 100."))
-	for {
-		resp, err := iter.Next()
-		if err == iterator.Done {
-			break
-		}
-		if err != nil {
-			log.Fatal(err)
-		}
-		_ = resp
 	}
 }

--- a/genai/client_test.go
+++ b/genai/client_test.go
@@ -58,13 +58,13 @@ func TestLive(t *testing.T) {
 			t.Fatal(err)
 		}
 		got := responseString(resp)
-		checkMatch(t, got, `[0-9]+ (cm|inches)`)
+		checkMatch(t, got, `[0-9]+ (cm|centimeters|inches)`)
 	})
 
 	t.Run("streaming", func(t *testing.T) {
 		iter := model.GenerateContentStream(ctx, Text("Are you hungry?"))
 		got := responsesString(t, iter)
-		checkMatch(t, got, `(don't|do\s+not) (have|possess) .*(a .* needs|body|the ability)`)
+		checkMatch(t, got, `(don't|do\s+not|not capable) (have|possess|experiencing) .*(a .* needs|body|sensations|the ability)`)
 	})
 
 	t.Run("chat", func(t *testing.T) {
@@ -448,5 +448,27 @@ func TestMatchString(t *testing.T) {
 		if !re.MatchString(test.in) {
 			t.Errorf("%q doesn't match %q", test.re, test.in)
 		}
+	}
+}
+
+func TestStreamCount(t *testing.T) {
+	ctx := context.Background()
+	client, err := NewClient(ctx, option.WithAPIKey(*apiKey))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+	model := client.GenerativeModel(*modelName)
+
+	iter := model.GenerateContentStream(ctx, Text("count 1 to 100."))
+	for {
+		resp, err := iter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			log.Fatal(err)
+		}
+		_ = resp
 	}
 }

--- a/genai/example_test.go
+++ b/genai/example_test.go
@@ -215,8 +215,10 @@ func ExampleClient_ListModels() {
 
 func printResponse(resp *genai.GenerateContentResponse) {
 	for _, cand := range resp.Candidates {
-		for _, part := range cand.Content.Parts {
-			fmt.Println(part)
+		if cand.Content != nil {
+			for _, part := range cand.Content.Parts {
+				fmt.Println(part)
+			}
 		}
 	}
 	fmt.Println("---")

